src/layers/core/geojson-layer/geojson-layer.js

```diff
@@ -80,20 +80,19 @@ export default class GeoJsonLayer extends CompositeLayer {
   renderLayers() {
     const {features} = this.state;
     const {pointFeatures, lineFeatures, polygonFeatures, polygonOutlineFeatures} = features;
-    const {getColor, getFillColor, getRadius, getWidth, getElevation} = this.props;
+    const {getColor, getFillColor, getRadius, getWidth, getElevation, updateTriggers} = this.props;
     const {id, stroked, filled, extruded, wireframe} = this.props;
 
     let {} = this.props;
     const drawPoints = pointFeatures && pointFeatures.length > 0;
     const drawLines = lineFeatures && lineFeatures.length > 0;
-    const drawPolygons = stroked && polygonOutlineFeatures && polygonOutlineFeatures.length > 0;
-    const fillPolygons = filled && polygonFeatures && polygonFeatures.length > 0;
+    const drawPolygons = polygonOutlineFeatures && polygonOutlineFeatures.length > 0;
 
     const onHover = this._onHoverSubLayer.bind(this);
     const onClick = this._onClickSubLayer.bind(this);
 
     // Filled Polygon Layer
-    const polygonFillLayer = fillPolygons && new SolidPolygonLayer(Object.assign({}, this.props, {
+    const polygonFillLayer = filled && drawPolygons && new SolidPolygonLayer(Object.assign({}, this.props, {
```

Comment:
This doesn't look right to me - `drawPolygons` is derived from `polygonOutlineFeatures`, not `polygonFeatures`... See discussion below, maybe we can drop the length check on the data.